components/server/ee/src/workspace/gitpod-server-impl.ts

```diff
@@ -1435,8 +1435,25 @@ export class GitpodServerEEImpl extends GitpodServerImpl {
         }
         const projects = await this.projectsService.getProjectsByCloneUrls(repositories.map(r => r.cloneUrl));
 
-        const cloneUrlsInUse = new Set(projects.map(p => p.cloneUrl));
-        repositories.forEach(r => { r.inUse = cloneUrlsInUse.has(r.cloneUrl) });
+        const cloneUrlToProject = new Map(projects.map(p => [p.cloneUrl, p]));
+
+        for (const repo of repositories) {
+            const p = cloneUrlToProject.get(repo.cloneUrl);
```

Comment:
Nit: Maybe the Map is a bit overkill here

```suggestion
            const p = projects.find(p => p.cloneUrl === repo.cloneUrl)!;
```